services/notifications/pkg/service/service.go

```diff
@@ -87,52 +97,101 @@ func (s eventsNotifier) Run() error {
 	}
 }
 
-func (s eventsNotifier) render(template email.MessageTemplate, values map[string]interface{}) (string, string, error) {
-	// The locate have to come from the user setting
-	return email.RenderEmailTemplate(template, "en", s.emailTemplatePath, s.translationPath, values)
+// recipient represent the already rendered message including the user id opaqueID
+type recipient struct {
+	opaqueID string
+	subject  string
+	msg      string
 }
 
-func (s eventsNotifier) send(ctx context.Context, u *user.UserId, g *group.GroupId, msg, subj, sender string) error {
-	if u != nil {
-		return s.channel.SendMessage(ctx, []string{u.GetOpaqueId()}, msg, subj, sender)
+func (s eventsNotifier) render(ctx context.Context, template email.MessageTemplate,
+	granteeFieldName string, fields map[string]interface{}, granteeList []*user.UserId) ([]*recipient, error) {
+	// Render the Email Template for each user
+	recipientList := make([]*recipient, len(granteeList))
```

Comment:
does this need to be a `*recipient`? Can't it just be a `recipient`?

Reply:
Yes we can

Reply:
done